client/web/src/enterprise/executors/secrets/RemoveSecretModal.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useCallback } from 'react'
+
+import { ErrorAlert } from '@sourcegraph/branded/src/components/alerts'
+import { logger } from '@sourcegraph/common'
+import { Button, H3, Modal } from '@sourcegraph/wildcard'
+
+import { LoaderButton } from '../../../components/LoaderButton'
+import { ExecutorSecretFields } from '../../../graphql-operations'
+
+import { useDeleteExecutorSecret } from './backend'
+
+export interface RemoveSecretModalProps {
+    secret: ExecutorSecretFields
+
+    onCancel: () => void
+    afterDelete: () => void
+}
+
+export const RemoveSecretModal: React.FunctionComponent<React.PropsWithChildren<RemoveSecretModalProps>> = ({
+    secret,
+    onCancel,
+    afterDelete,
+}) => {
+    const labelId = 'removeSecret'
+
+    const [deleteExecutorSecret, { loading, error }] = useDeleteExecutorSecret()
+
+    const onDelete = useCallback<React.MouseEventHandler>(
+        async event => {
+            event.preventDefault()
+
+            try {
+                await deleteExecutorSecret({ variables: { id: secret.id, scope: secret.scope } })
+
+                afterDelete()
+            } catch (error) {
+                // Non-request error. API errors will be available under `error` above.
+                logger.error(error)
+            }
+        },
+        [afterDelete, secret.id, secret.scope, deleteExecutorSecret]
+    )
+
+    return (
+        <Modal onDismiss={onCancel} aria-labelledby={labelId}>
+            <H3 id={labelId}>Remove secret {secret.key}?</H3>
```

Comment:
```suggestion
            <H3 id={labelId}>Executor secret: {secret.key}</H3>
```

Might be more our style, looking at the Remove Credentials modal for inspiration.